src/Compilers/CSharp/Test/Emit/Emit/DynamicAnalysis/DynamicInstrumentationTests.cs

```diff
@@ -1169,8 +1187,186 @@ .locals init (bool[] V_0)
+  IL_0005:  ldc.i4.1
+  IL_0006:  add
+  IL_0007:  newarr     ""bool[]""
+  IL_000c:  stsfld     ""bool[][] <PrivateImplementationDetails>.PayloadRoot0""
+  IL_0011:  ldtoken    ""<PrivateImplementationDetails>""
+  IL_0016:  call       ""System.Type System.Type.GetTypeFromHandle(System.RuntimeTypeHandle)""
+  IL_001b:  call       ""System.Reflection.TypeInfo System.Reflection.IntrospectionExtensions.GetTypeInfo(System.Type)""
+  IL_0020:  callvirt   ""System.Reflection.Module System.Reflection.MemberInfo.Module.get""
+  IL_0025:  callvirt   ""System.Guid System.Reflection.Module.ModuleVersionId.get""
+  IL_002a:  stsfld     ""System.Guid <PrivateImplementationDetails>.MVID""
+  IL_002f:  ret
+}";
+
             CompilationVerifier verifier = CompileWithNoFramework(source + InstrumentationHelperSource, emitOptions: EmitOptions.Default.WithInstrument("Test.Flag"));
             verifier.VerifyIL("Program.TestMain", expectedTestMainIL);
+            verifier.VerifyIL(".cctor", expectedPIDStaticConstructorIL);
+        }
+
+        [Fact]
+        public void MissingMethodNeededForAnalysis()
+        {
+            string source = @"
+namespace System
+{
+    public class Object { }  
+    public struct Int32 { }  
+    public struct Boolean { }  
+    public class String { }  
+    public class Exception { }  
+    public class ValueType { }  
+    public class Enum { }  
+    public struct Void { }  
+    public class Guid { }
+    public struct RuntimeTypeHandle { }
+
+    public class Type
+    {
+        public static Type GetTypeFromHandle(RuntimeTypeHandle handle) { return new Type(); }
+    }
+}
+
+public class Console
+{
+    public static void WriteLine(string s) { }
+    public static void WriteLine(int i) { }
+    public static void WriteLine(bool b) { }
+}
+
+namespace System.Reflection
+{
+    public class MemberInfo
+    {
+        public virtual Module Module => new Module();
+    }
+
+    public class TypeInfo : MemberInfo
+    {
+    }
+
+    public class Module
+    {
+        public virtual Guid ModuleVersionId => new Guid();
+    }
+
+    public static class IntrospectionExtensions
+    {
+        // public static TypeInfo GetTypeInfo(System.Type t) { return new TypeInfo(); }
+    }
+}
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        TestMain();
+        Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload();
+    }
+
+    static int TestMain()
+    {
+        return 3;
+    }
+}
+";
+
+            try
+            {
+                CompilationVerifier verifier = CompileWithNoFramework(source + InstrumentationHelperSource, emitOptions: EmitOptions.Default.WithInstrument("Test.Flag"));
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember missing)
+            {
+                Assert.True(missing.Diagnostic.ToString().Contains("IntrospectionExtensions.GetTypeInfo"));
+                return;                
+            }
```

Comment:
I'm a little worried about this `catch` because I'm not sure if the exception comes from the test framework or the compiler, and the compiler should not throw in such a case.

Is it possible to use this tests pattern to check the diagnostics instead?

```C#

            var comp = CreateCompilationWithMscorlib(source);
            comp.VerifyDiagnostics(
                // (6,9): error CS8058: Feature 'tuples' is experimental and unsupported; use '/features:tuples' to enable.
                //         (int, int) x = (1, 1);
                Diagnostic(ErrorCode.ERR_FeatureIsExperimental, "(int, int)").WithArguments("tuples", "tuples").WithLocation(6, 9),
                // (6,24): error CS8058: Feature 'tuples' is experimental and unsupported; use '/features:tuples' to enable.
                //         (int, int) x = (1, 1);
                Diagnostic(ErrorCode.ERR_FeatureIsExperimental, "(1, 1)").WithArguments("tuples", "tuples").WithLocation(6, 24)
                );
```

Run it once with `comp.VerifyDiagnostics();` and the output window will give you what you can copy and paste into the parameters.